### PR TITLE
ARGV: Deprecate ARGV.collect_build_args

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#5730

### CODE REVIEW

File: Library/Homebrew/cli/args.rb [hunk 1]
```diff
@@ -167,6 +180,19 @@ def devel
         (args_parsed && devel?) || cmdline_args.include?("--devel")
       end
 
+      def build_universal
+        (args_parsed && universal?) || cmdline_args.include?("--universal")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cli/args.rb]: Might be nice to split this (and all similar methods) into a `return` on each set of `()`

Comment (GauthamGoli) [reply Library/Homebrew/cli/args.rb]: I'm not sure I understand, could you please suggest the code?

Comment (MikeMcQuaid) [inline Library/Homebrew/cli/args.rb]:
```suggestion
        return true if args_parsed && universal?
        
        cmdline_args.include?("--universal")
```

File: Library/Homebrew/cli/args.rb [hunk 2]
```diff
@@ -167,6 +180,19 @@ def devel
...
+      end
+
+      def build_bottle
+        (args_parsed && build_bottle?) || cmdline_args.include?("--build-bottle")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cli/args.rb]:
```suggestion
        return true if args_parsed && build_bottle?
        
        cmdline_args.include?("--build-bottle")
```

File: Library/Homebrew/cli/args.rb [hunk 3]
```diff
@@ -167,6 +180,19 @@ def devel
...
+      end
+
+      def build_from_source
+        (args_parsed && (build_from_source? || s?)) || cmdline_args.include?("--build-from-source") ||
+          cmdline_args.include?("-s")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cli/args.rb]:
```suggestion
        return true if args_parsed && (build_from_source? || s?) 
        
        cmdline_args.include?("--build-from-source") || cmdline_args.include?("-s")
```

Comment (GauthamGoli) [reply Library/Homebrew/cli/args.rb]: Seems clearer and more readable. Thanks!

**Pull request merged**